ddl/column.go

```diff
@@ -316,7 +317,14 @@ func (d *ddl) addTableColumn(t table.Table, columnInfo *model.ColumnInfo, reorgI
 
 	for {
 		startTime := time.Now()
-		handles, err := d.getSnapshotRows(t, version, seekHandle)
+		handles = handles[:0]
+		err = d.getSnapshotRows(t, version, seekHandle, func(h int64, rowKey kv.Key, rawRecord []byte) (bool, error) {
+			handles = append(handles, h)
+			if len(handles) == defaultBatchCnt {
```

Comment:
`>=`  just make me "think" it's safer, because the code won't go to dead loop when any small mistake happens, makes `len(xxx) > yyy` already...